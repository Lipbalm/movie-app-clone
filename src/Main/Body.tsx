import React, { FC, useEffect } from "react";
import { RouterHomeContainer, CarouselContainer, BodyWrapper } from "../Style";
import { getGenreIds } from "../Modules/apis";
import CardWidget from "../Components/WidgetComponents/CardWidget";

const Body: FC = () => {
  useEffect(() => {
    const fetchData = async () => {
      await getGenreIds("en-US");
    };
    fetchData();
  }, []);

  return (
    <RouterHomeContainer>
      <CarouselContainer></CarouselContainer>
      <BodyWrapper>
        <CardWidget key="CardWidget1"></CardWidget>
        <CardWidget key="CardWidget2"></CardWidget>
        <CardWidget key="CardWidget3"></CardWidget>
      </BodyWrapper>
    </RouterHomeContainer>
  );
};

export default Body;
