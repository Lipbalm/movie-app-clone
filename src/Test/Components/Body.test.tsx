import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Router } from "react-router-dom";
import Body from "../../Main/Body";
import { createMemoryHistory } from "history";

const setup = () => {
  const history = createMemoryHistory();
  return {
    ...render(
      <Router history={history}>
        <Body />
      </Router>
    )
  };
};
describe("<Body />", () => {
  it("snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it("CardWidget HeadLink Click", () => {
    const { getByText, container } = setup();
    const cardWidgetHeadLink = getByText("더보기");

    expect(container.innerHTML).toMatch("2020년");

    fireEvent.click(cardWidgetHeadLink);

    expect(container.innerHTML).toMatch(/ThemePage/);
  });
});
