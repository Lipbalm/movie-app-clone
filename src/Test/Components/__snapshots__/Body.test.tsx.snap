// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> snapshot 1`] = `
<div>
  <div
    class="sc-htpNat LkrjQ"
  >
    <div
      class="sc-bxivhb czzmdl"
    />
    <div
      class="sc-ifAKCX cktARQ"
    >
      <div
        class="sc-EHOje evmUqh"
      >
        <h2
          class="sc-bZQynM jdkaLI"
        >
          2020년 1분기 기대신작 애니
          <a
            class="sc-gzVnrw fMaZf"
            href="/theme"
          >
            더보기
          </a>
        </h2>
        <div
          class="sc-htoDjs bZzhDu"
        >
          <div
            class="sc-dnqmqq dObvuR"
          >
            <div
              class="sc-iwsKbI fdCiff"
              data-testid="card"
            >
              <div
                class="sc-gqjmRU iwysUm"
              >
                <img
                  alt="no image"
                  class="sc-VigVT gfvvmf"
                  src="https://image.tmdb.org/t/p/w500//2CAL2433ZeIihfX1Hb2139CX0pW.jpg"
                />
              </div>
              <div
                class="sc-jTzLTM iJvBBd"
              >
                <span
                  class="sc-fjdhpX fYgqRV"
                  data-testid="genre"
                >
                  35/18/10749
                </span>
                <a
                  class="sc-jzJRlG dmluJx"
                  data-testid="title"
                >
                  Dilwale Dulhania Le Jayenge
                </a>
              </div>
            </div>
            <div
              class="sc-iwsKbI buVUVn"
              data-testid="card"
            >
              <div
                class="sc-gqjmRU iwysUm"
              >
                <img
                  alt="no image"
                  class="sc-VigVT gfvvmf"
                  src="https://image.tmdb.org/t/p/w500//9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg"
                />
              </div>
              <div
                class="sc-jTzLTM iJvBBd"
              >
                <span
                  class="sc-fjdhpX fYgqRV"
                  data-testid="genre"
                >
                  80/18
                </span>
                <a
                  class="sc-jzJRlG dmluJx"
                  data-testid="title"
                >
                  The Shawshank Redemption
                </a>
              </div>
            </div>
            <div
              class="sc-iwsKbI buVUVn"
              data-testid="card"
            >
              <div
                class="sc-gqjmRU iwysUm"
              >
                <img
                  alt="no image"
                  class="sc-VigVT gfvvmf"
                  src="https://image.tmdb.org/t/p/w500//rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg"
                />
              </div>
              <div
                class="sc-jTzLTM iJvBBd"
              >
                <span
                  class="sc-fjdhpX fYgqRV"
                  data-testid="genre"
                >
                  18/80
                </span>
                <a
                  class="sc-jzJRlG dmluJx"
                  data-testid="title"
                >
                  The Godfather
                </a>
              </div>
            </div>
            <div
              class="sc-iwsKbI buVUVn"
              data-testid="card"
            >
              <div
                class="sc-gqjmRU iwysUm"
              >
                <img
                  alt="no image"
                  class="sc-VigVT gfvvmf"
                  src="https://image.tmdb.org/t/p/w500//yPisjyLweCl1tbgwgtzBCNCBle.jpg"
                />
              </div>
              <div
                class="sc-jTzLTM iJvBBd"
              >
                <span
                  class="sc-fjdhpX fYgqRV"
                  data-testid="genre"
                >
                  18/36/10752
                </span>
                <a
                  class="sc-jzJRlG dmluJx"
                  data-testid="title"
                >
                  Schindler's List
                </a>
              </div>
            </div>
            <div
              class="sc-iwsKbI buVUVn"
              data-testid="card"
            >
              <div
                class="sc-gqjmRU iwysUm"
              >
                <img
                  alt="no image"
                  class="sc-VigVT gfvvmf"
                  src="https://image.tmdb.org/t/p/w500//7IiTTgloJzvGI1TAYymCfbfl3vT.jpg"
                />
              </div>
              <div
                class="sc-jTzLTM iJvBBd"
              >
                <span
                  class="sc-fjdhpX fYgqRV"
                  data-testid="genre"
                >
                  35/18/53
                </span>
                <a
                  class="sc-jzJRlG dmluJx"
                  data-testid="title"
                >
                  Parasite
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
